src/kv/RocksDBStore.cc

```diff
@@ -1224,15 +1224,18 @@ void RocksDBStore::compact_range_async(const string& start, const string& end)
       // dup; no-op
       return;
     }
-    if (p->first <= end && p->first > start) {
-      // merge with existing range to the right
-      compact_queue.push_back(make_pair(start, p->second));
+    if (start <= p->first && p->first <= end) {
+      // new region crosses start of existing range
+      // select right bound that is bigger
+      compact_queue.push_back(make_pair(start, end > p->second ? end : p->second));
```

Comment:
why not just use `interval_set`, it is able to cover all the cases and less error prone.

Reply:
Good idea. I will change it if I refactor compact_queue.